pkg/controller/scheduler.go

```diff
@@ -718,32 +753,43 @@ func (c *Controller) shouldSkipAnalysis(canary *flaggerv1.Canary, canaryControll
 	c.recorder.SetWeight(canary, primaryWeight, canaryWeight)
 
 	// copy spec and configs from canary to primary
-	c.recordEventInfof(canary, "Copying %s.%s template spec to %s-primary.%s",
-		canary.Spec.TargetRef.Name, canary.Namespace, canary.Spec.TargetRef.Name, canary.Namespace)
-	if err := canaryController.Promote(canary); err != nil {
-		c.recordEventWarningf(canary, "%v", err)
-		return true
+	if canary.Status.Phase != flaggerv1.CanaryPhasePromoting {
```

Comment:
When `skipAnalysis` is enabled, it means that we halt the entire state machine, promote the canary, scale down the canary deployment and mark it as `Succeeded`, this changes that and relies on the state machine to continue and mark it as succeeded, which is not we want to do.